packages/compose/src/hooks/use-focus-outside/index.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * External dependencies
+ */
+import type {
+	FocusEventHandler,
+	EventHandler,
+	MouseEventHandler,
+	TouchEventHandler,
+	FocusEvent,
+	MouseEvent,
+	TouchEvent,
+} from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Input types which are classified as button types, for use in considering
+ * whether element is a (focus-normalized) button.
+ */
+const INPUT_BUTTON_TYPES = [ 'button', 'submit' ];
+
+type FocusNormalizedButton =
+	| HTMLButtonElement
+	| HTMLLinkElement
+	| HTMLInputElement;
+
+/**
+ * Returns true if the given element is a button element subject to focus
+ * normalization, or false otherwise.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Clicking_and_focus
+ *
+ * @param  eventTarget The target from a mouse or touch event.
```

Comment:
Missing the param type?

```suggestion
 * @param  {EventTarget} eventTarget The target from a mouse or touch event.
```

Reply:
In the past, I've usually avoided having types in JSDocs in TypeScript files, as the code itself is already typed. The only exceptions have been for when the JSDoc types where necessary for automatic doc generation.

Adding types here would also require us to add back a lot of JSDoc extra type def (see the code deletions in this file)

Reply:
I generally agree, just wanted to pick your thoughts 👍 Happy to keep it as you suggested.